pipelines/build/common/openjdk_build_pipeline.groovy

```diff
@@ -1436,8 +1436,11 @@ class Build {
                                             """.stripIndent(), returnStdout: true, returnStatus: false).replaceAll('\n', '')
             } else {
                 context.println "Windows detected - running bat to generate SHA256 sums in writeMetadata"
-                hash = context.bat(script: "sha256sum ${file} | cut -f1 -d' '") // .replaceAll('\n', '')
+                String hash_stdout = context.bat(script: "sha256sum ${file}", returnStdout: true, returnStatus: false)
+                // Windows batch returns stdout of <command invoked>\r\n<sha256sum output>\r\n
+                hash = hash_stdout.split('\r').last().replaceAll('\n', '').split(' ').first()
```

Comment:
I'm a bit confused - what do you mean by "Windows batch" in this comment? I don't recall seeing a situation from the windows command line where sha256sum - which I assume is the cygwin one - would behave like that. If it's more subtle than that then we should adjust this comment.

Reply:
Ref my [other comment](https://github.com/adoptium/ci-jenkins-pipelines/pull/1208/files#r2027410485) I see this is specific to the pipeline `bat` operation. You could prevent it by using `@sha256sum` which might keep the operation a bit simpler.

```suggestion
                String hash_stdout = context.bat(script: "@sha256sum ${file}", returnStdout: true, returnStatus: false)
```

Or if you don't want to do that can we change the comment please?
```suggestion
                // "bat" in pipelines returns stdout of <command invoked>\r\n<sha256sum output>\r\n
```

Reply:
i'll give that a try, thanks

Reply:
Test build https://ci.adoptium.net/job/build-scripts/job/jobs/job/jdk21u/job/jdk21u-windows-x64-temurin/269/

Reply:
tidier yes, need to split, but should work better yes, thanks